modules/webgl/src/adapter/resources/webgl-transform-feedback.ts

```diff
@@ -0,0 +1,253 @@
+import type {Device, ShaderLayout, TransformFeedbackProps} from '@luma.gl/core';
+import {log, isObjectEmpty, TransformFeedback, Buffer} from '@luma.gl/core';
+import {GL} from '@luma.gl/constants';
+import {WebGLDevice} from '../webgl-device';
+import {WEBGLBuffer, isWebGL2} from '../..';
+
+/** For bindRange */
+type BufferRange = {
+  buffer: WEBGLBuffer;
+  byteOffset?: number;
+  byteLength?: number;
+};
+
+export class WEBGLTransformFeedback extends TransformFeedback {
+  get [Symbol.toStringTag](): string {
+    return 'TransformFeedback';
+  }
+
+  readonly device: WebGLDevice;
+  readonly gl2: WebGL2RenderingContext;
+  readonly handle: WebGLTransformFeedback;
+
+  /**
+   * NOTE: The Model already has this information while drawing, but
+   * TransformFeedback currently needs it internally, to look up
+   * varying information outside of a draw() call.
+   */
+  readonly layout: ShaderLayout;
+  buffers: Record<string, BufferRange> = {};
+  unusedBuffers: Record<string, Buffer> = {};
+  // NOTE: The `bindOnUse` flag is a major workaround:
+  // See https://github.com/KhronosGroup/WebGL/issues/2346
+  bindOnUse = true;
+  private _bound: boolean = false;
+
+  constructor(device: WebGLDevice, props: TransformFeedbackProps) {
+    super(device, props);
+
+    device.assertWebGL2();
+    this.device = device;
+    this.gl2 = device.gl2;
+    this.handle = this.props.handle || this.gl2.createTransformFeedback();
+    this.layout = this.props.layout;
+
+    // TODO(v9): Should this logic exist in the constructor?
+    try {
+      this.gl2.bindTransformFeedback(GL.TRANSFORM_FEEDBACK, this.handle);
+      for (const [locationOrName, bufferOrRange] of Object.entries(this.props.buffers)) {
+        const location = this._getVaryingIndex(locationOrName);
+        const range = this._getBufferRange(bufferOrRange as WEBGLBuffer | BufferRange);
+        if (location < 0) {
+          this.unusedBuffers[locationOrName] = range.buffer;
+          log.warn(`${this.id} unused varying buffer ${locationOrName}`)();
+          return this;
+        }
+        this.buffers[location] = range;
+      }
+    } finally {
+      this.gl2.bindTransformFeedback(GL.TRANSFORM_FEEDBACK, null);
+    }
+
+    // Need to avoid chrome bug where buffer that is already bound to a different target
+    // cannot be bound to 'TRANSFORM_FEEDBACK_BUFFER' target.
+    if (!this.bindOnUse) {
+      this._bindBuffers();
+    }
+
+    Object.seal(this);
+  }
+
+  static override isSupported(device: Device | WebGLRenderingContext): boolean {
+    const webglDevice = WebGLDevice.attach(device);
+    return isWebGL2(webglDevice.gl);
+  }
+
+  override destroy(): void {
+    this.gl2.deleteTransformFeedback(this.handle);
+    super.destroy();
+  }
+
+  begin(primitiveMode = GL.POINTS): this {
```

Comment:
I suspect these should override some methods in the core class, otherwise they cannot be called by an application without importing webgl module. That also means that we should replace `GL.POINTS` with e.g. a string constant uniion
```suggestion
  begin(primitiveMode = 'points' | ...): this {
```

Reply:
These should have a JSDOC comment, in the parent class if they are overridden.

I have been moving away from returning `this` and just return `void`. Chaining has not been very useful.